src/dotty/tools/dotc/reporting/Reporter.scala

```diff
@@ -210,6 +210,8 @@ abstract class Reporter {
   var warningCount = 0
   def hasErrors = errorCount > 0
   def hasWarnings = warningCount > 0
+  private var errors: List[Error] = Nil
+  def allErrors = errors
```

Comment:
I am not super-happy with recording all errors here. That should be the job of individual reporters, not of the base class `Reporter`. I believe it would be better to install a particular reporter that stores the errors in the test framework.